File: 백준/Silver/2910. 빈도 정렬/빈도 정렬.js
const fs = require('fs');
let input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

let [n,c] = input.shift().split(' ').map(Number);
let arr = input[0].split(' ').map(Number)

let result = [];
const freqMap = new Map();

for (let i=0; i<n; i++) {
  freqMap.set(arr[i],(freqMap.get(arr[i])||0)+1);
}

const sortedMap = [...freqMap].sort((a,b) => b[1]-a[1]);
sortedMap.map((info) => {
  let [k,v] = [info[0], info[1]];
  let tmp = 0;
  while (tmp < v) {
    result.push(k);
    tmp += 1;
  }
})

for (let j=0; j<result.length; j++) {
  process.stdout.write(result[j] + ' ')
}

File: 백준/Silver/2910. 빈도 정렬/빈도 정렬.py
import sys
input = sys.stdin.readline

n,c = map(int,input().split())
arr = list(map(int,input().split()))
result = []

f_dict = {}
for a in arr:
    if a not in f_dict:
        f_dict[a] = 1
    else:
        f_dict[a] += 1

sorted_f = list(f_dict.items())
sorted_f.sort(key=lambda x:x[1] ,reverse=True)

for a,b in sorted_f:
    tmp = 0
    while tmp < b:
        result.append(a)
        tmp += 1
        
print(*result)